### DISCUSSION TITLE
Logs

### DISCUSSION THREAD

**Discussion by filipedeschamps** [BrasilAPI/BrasilAPI#248] [Show and tell]
Algo para vermos depois: https://zeit.co/blog/log-drains

**Comment by lucianopf** [comment]
Caraaaaca, isso é sensacional, não tinha visto esse lançamento não! 😱 

No momento os conectores criados são so pra Datadog (que é sensacional, mas acho meio caro) e o LogDna que nunca usei mas parece legal.

Dei uma boa lida em como criar um log-drain aqui https://zeit.co/docs/api#integrations/log-drains/create-a-log-drain

E como foram escritos os conectores pro Datadog e pro LogDna, e aparentemente não é muito complexo de se criar um (eu já criei umas integrações lá e de fato não é algo mega complexo não), só precisaremos decidir onde iremos jogar os logs de forma que seja barato, tranquilo de consumir, e que de certa forma permita múltiplos usuários 🤔 😬

**Comment by filipedeschamps** [comment]
Vou abrir o coração: tenho vontade de usar o Datadog só por conta do logo 😂

**Comment by lucianopf** [comment]
Ele é muito bom pra APM e métricas em geral mas como todas ferramentas de métricas tem os pontos negativos, o dele na minha opinião é o sistema de logs que é bom mas perde pra simplicidade tipo do Logentries :grimacing: 

Eu tô muito de olho nesse aqui do TJ https://apex.sh/logs/

Ta parecendo animal! :grimacing:

**Comment by filipedeschamps** [comment]
Ahhhh boaaa, é verdade!!! Topo total testar! Me inscrevi no mailling dele 👍

**Comment by paulogdm** [comment]
Logflare é a alternativa recomendada para projetos indies. Muito bom o custo vs benefício e extremamente eficiente.

**Comment by lucianopf** [comment]
Que massa que vc falou disso @paulogdm !!

Inclusive to usando no proxier.now.sh que é usado no `cep-promise` 😬 

Comecei a coletar os logs sábado e agora já tem uns 60k 😂 

@filipedeschamps vou te mandar o invite pra vc dar uma olhadinha!

@paulogdm na verdade não consegui entender muito o modelo de pricing deles, é tudo meio obscuro, pelo que entendi é free, tem alguma coisa que to ignorando lá? 🤔

**Comment by paulogdm** [comment]
Rapaz somos dois. Não dá pra entender nada, só que é barato... Conversando com o desenvolvedor um pouco ainda não está claro que direção eles vão seguir...

**Comment by lucianopf** [comment]
Boa, que bom que não sou o único perdido hahaha, ao menos é algo que sabem que precisam melhorar 😬 

O que mais gostei foi a possibilidade de abrir o logset usando o Google Data Studio, sério, dá pra fazer muita coisa!!

@filipedeschamps chegou a dar uma olhadinha nesse `explore`? 👀

**Comment by devarthurribeiro** [comment]
Estou utilizando o https://logflare.app/ ele integra com o Google Data Studio. O plano grátis é de até 5,200,000 logs por mês.

[Exemplo](https://datastudio.google.com/u/0/reporting/1OiXkL1AOTJw4-dpudE1y_AGQc_mThGm-?s=gIAFsUMbDS4) do logflare mais Google Data Studio.
